spt/OrangeBox/scripts/framebulk_handler.cpp

```diff
@@ -0,0 +1,242 @@
+#include "stdafx.h"
+#include "framebulk_handler.hpp"
+#include "..\..\utils\string_parsing.hpp"
+
+namespace scripts
+{
+	typedef void(*CommandCallback) (FrameBulkInfo& frameBulkInfo);
+	std::vector<CommandCallback> frameBulkHandlers;
+
+	const std::string FIELD0_FILLED = "s**ljdbcgu";
+	const std::string FIELD1_FILLED = "flrbud";
+	const std::string FIELD2_FILLED = "jdu12r";
+	const std::string EMPTY_FIELD = "-";
+	const char WILDCARD = '*';
+	const char DELIMITER = '|';
+
+	const auto STRAFE = std::pair<int, int>(0, 0);
+	const auto STRAFE_TYPE = std::pair<int, int>(0, 1);
+	const auto JUMP_TYPE = std::pair<int, int>(0, 2);
+	const auto LGAGST = std::pair<int, int>(0, 3);
+	const auto AUTOJUMP = std::pair<int, int>(0, 4);
+	const auto DUCKSPAM = std::pair<int, int>(0, 5);
+	const auto JUMPBUG = std::pair<int, int>(0, 6);
+	const auto DUCK_BEFORE_COLLISION = std::pair<int, int>(0, 7);
+	const auto DUCK_BEFORE_GROUND = std::pair<int, int>(0, 8);
+	const auto USE_SPAM = std::pair<int, int>(0, 9);
+
+	const auto FORWARD = std::pair<int, int>(1, 0);
+	const auto LEFT = std::pair<int, int>(1, 1);
+	const auto RIGHT = std::pair<int, int>(1, 2);
+	const auto BACK = std::pair<int, int>(1, 3);
+	const auto UP = std::pair<int, int>(1, 4);
+	const auto DOWN = std::pair<int, int>(1, 5);
+
+	const auto JUMP = std::pair<int, int>(2, 0);
+	const auto DUCK = std::pair<int, int>(2, 1);
+	const auto USE = std::pair<int, int>(2, 2);
+	const auto ATTACK1 = std::pair<int, int>(2, 3);
+	const auto ATTACK2 = std::pair<int, int>(2, 4);
+	const auto RELOAD = std::pair<int, int>(2, 5);
+
+	const auto YAW_KEY = std::pair<int, int>(3, 0);
+	const auto PITCH_KEY = std::pair<int, int>(4, 0);
+	const auto TICKS = std::pair<int, int>(5, 0);
+	const auto COMMANDS = std::pair<int, int>(6, 0);
+
+	void Field1(FrameBulkInfo& frameBulkInfo)
```

Comment:
why would constant variables get allocated multiple times?

Reply:
I think they should be static since they are declared outside the function???

Reply:
Every time you pass a `std::string` somewhere by value, and every time you add `std::string`s with a `+` there are new buffers being allocated.

Reply:
Even if you do `static const std::string = "whatever";` it will allocate it on the heap and initialize whenever SPT gets loaded the first time. `std::string` is a heap-allocated string.

Reply:
Tried to add const references where possible and got rid of some of the += string operators